File: Ahorn/entities/AnonCloud.jl
module AnonhelperAnonCloud

using ..Ahorn, Maple

@mapdef Entity "Anonhelper/AnonCloud" AnonCloud(x::Integer, y::Integer, small::Bool=false, pink::Bool=false)


const placements = Ahorn.PlacementDict(
    "AnonCloud (Anonhelper)" => Ahorn.EntityPlacement(
	AnonCloud,
	"point"
   )
)

const sprites = Dict{Tuple{Bool, Bool}, String}(
	(false, false) => "objects/Clouds/whitecloud00",
	(true, false) => "objects/Clouds/pinkcloud00",
	(false, true) => "objects/Clouds/whitecloudRemix00",
	(true, true) => "objects/Clouds/pinkcloudRemix00",
)

function Ahorn.selection(entity::AnonCloud)
    pink = get(entity.data, "pink", false)
    small = get(entity.data, "small", false)
    sprite = sprites[(pink, small)]
    x, y = Ahorn.position(entity)
    return Ahorn.getSpriteRectangle(sprite, x, y)
end

function Ahorn.render(ctx::Ahorn.Cairo.CairoContext, entity::AnonCloud, room::Maple.Room)
	pink = get(entity.data, "pink", false)
	small = get(entity.data, "small", false)
	sprite = sprites[(pink, small)]
   Ahorn.drawSprite(ctx, sprite, 0, 0)
end

end

File: Loenn/entities/anon_cloud.lua
local utils = require("utils")

local anonCloud = {}

anonCloud.name = "Anonhelper/AnonCloud"
anonCloud.depth = 0
anonCloud.placements = {
    {
        name = "normal",
        data = {
            pink = false,
            small = false
        }
    },
    {
        name = "fragile",
        data = {
            pink = true,
            small = false
        }
    }
}

function anonCloud.texture(room, entity)
    local small = entity.small
    local fragile = entity.pink

    if small then
        if fragile then
            return "objects/AnonHelper/clouds/pinkcloudRemix00"
        else
            return "objects/AnonHelper/clouds/whitecloudRemix00"
        end
    else
        if fragile then
            return "objects/AnonHelper/clouds/pinkcloud00"
        else
            return "objects/AnonHelper/clouds/whitecloud00"
        end
    end
end

function anonCloud.selection(room, entity)
	local x, y = entity.x or 0, entity.y or 0
    if entity.small then
        return utils.rectangle(x - 13, y - 6, 24, 15)
    else
        return utils.rectangle(x - 18, y - 6, 36, 15)
    end
end

return anonCloud